src/lib/Permission/Mapper/UsersWithPermissionInfoMapper.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\AdminUi\Permission\Mapper;
+
+use Ibexa\Contracts\AdminUi\Values\PermissionCheckContext;
+use Ibexa\Contracts\Core\Repository\PermissionResolver;
+use Ibexa\Contracts\Core\Repository\UserService;
+use Ibexa\Contracts\Core\Repository\Values\Content\Search\SearchResult;
+use Ibexa\Contracts\Core\Repository\Values\User\User;
+use Ibexa\Core\Repository\Values\User\UserReference;
+
+/**
+ * @phpstan-type TUserData array{
+ *     id: int,
+ *     name: string,
+ *     email: string,
+ * }
+ * @phpstan-type TPermissionInfoData array{
+ *     access: array<TUserData>,
+ *     no_access: array<TUserData>,
+ * }
+ */
+final class UsersWithPermissionInfoMapper
+{
+    private PermissionResolver $permissionResolver;
+
+    private UserService $userService;
+
+    public function __construct(
+        UserService $userService,
+        PermissionResolver $permissionResolver
+    ) {
+        $this->userService = $userService;
+        $this->permissionResolver = $permissionResolver;
+    }
+
+    /**
+     * @phpstan-return TPermissionInfoData
+     *
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\BadStateException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     */
+    public function mapSearchResults(
+        SearchResult $searchResult,
+        PermissionCheckContext $permissionContext,
+        string $module,
+        string $function
+    ): array {
+        $currentUserReference = $this->permissionResolver->getCurrentUserReference();
+
+        $results = $this->groupByPermissions($searchResult, $permissionContext, $module, $function);
+
+        $this->permissionResolver->setCurrentUserReference($currentUserReference);
```

Comment:
This instruction should be in catch { ... } finally { ... } block

Reply:
https://github.com/ibexa/admin-ui/pull/1372/commits/fc5337190391bd2dffa26b814c98a8958fd0bcec